File: Business Rule/Assemblies/Workforce Planning (WPP)/Workforce Planning (WPP)/WPP/service factory/service_factory.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Globalization;
using System.IO;
using System.Linq;
using Microsoft.CSharp;
using OneStream.Finance.Database;
using OneStream.Finance.Engine;
using OneStream.Shared.Common;
using OneStream.Shared.Database;
using OneStream.Shared.Engine;
using OneStream.Shared.Wcf;
using OneStream.Stage.Database;
using OneStream.Stage.Engine;
using OneStreamWorkspacesApi;
using OneStreamWorkspacesApi.V800;

namespace Workspace.__WsNamespacePrefix.__WsAssemblyName
{
	public class service_factory : IWsAssemblyServiceFactory
	{
        public IWsAssemblyServiceBase CreateWsAssemblyServiceInstance(SessionInfo si, BRGlobals brGlobals,
            DashboardWorkspace workspace, WsAssemblyServiceType wsAssemblyServiceType, string itemName)
        {
            try
            {
                switch (wsAssemblyServiceType)
                {
                    //case WsAssemblyServiceType.Component:
                    //    return new WsasComponent();

                    //case WsAssemblyServiceType.Dashboard:
                    //    return new WsasDashboard();

                    //case WsAssemblyServiceType.DataManagementStep:
                    //    return new WsasDataManagementStep();

                    //case WsAssemblyServiceType.DataSet:
                    //    return new WsasDataSet();

                    //case WsAssemblyServiceType.DynamicDashboards:
                    //    return new WsasDynamicDashboards();

                    //case WsAssemblyServiceType.DynamicCubeView:
                    //    return new WsasDynamicCubeView();

                    //case WsAssemblyServiceType.DynamicGrid:
                    //    return new WsasDynamicGrid();

                    //case WsAssemblyServiceType.FinanceCore:
                    //    return new WsasFinanceCore();

                    //case WsAssemblyServiceType.FinanceCustomCalculate:
                    //    return new WsasFinanceCustomCalculate();

                    //case WsAssemblyServiceType.FinanceGetDataCell:
                    //    return new WsasFinanceGetDataCell();

                    //case WsAssemblyServiceType.FinanceMemberLists:
                    //    return new WsasFinanceMemberLists();

                    //case WsAssemblyServiceType.SqlTableEditor:
                    //    return new WsasSqlTableEditor();

                    //case WsAssemblyServiceType.TableView:
                    //    return new WsasTableView();

                    //case WsAssemblyServiceType.XFBRString:
                    //    return new WsasXFBRString();

                    default:
                        return null;
                }
            }
            catch (Exception ex)
            {
                throw new XFException(si, ex);
            }
        }
	}
}

File: Business Rule/Assemblies/RES/RES/RES/service factory/service_factory.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Common
Imports System.Globalization
Imports System.IO
Imports System.Linq
Imports Microsoft.VisualBasic
Imports OneStream.Finance.Database
Imports OneStream.Finance.Engine
Imports OneStream.Shared.Common
Imports OneStream.Shared.Database
Imports OneStream.Shared.Engine
Imports OneStream.Shared.Wcf
Imports OneStream.Stage.Database
Imports OneStream.Stage.Engine
Imports OneStreamWorkspacesApi
Imports OneStreamWorkspacesApi.V800

Namespace Workspace.__WsNamespacePrefix.__WsAssemblyName
	Public Class service_factory
		Implements IWsAssemblyServiceFactory

        Public Function CreateWsAssemblyServiceInstance(ByVal si As SessionInfo, ByVal globals As BRGlobals, ByVal workspace As DashboardWorkspace, ByVal wsasType As WsAssemblyServiceType, ByVal itemName As String) As IWsAssemblyServiceBase Implements IWsAssemblyServiceFactory.CreateWsAssemblyServiceInstance
            Try
                Select Case wsasType
                    Case Is = WsAssemblyServiceType.Component
                        'Return New WsasComponent()

                    Case Is = WsAssemblyServiceType.Dashboard
                        'Return New WsasDashboard()

                    Case Is = WsAssemblyServiceType.DataManagementStep
                        'Return New WsasDataManagementStep()

                    Case Is = WsAssemblyServiceType.DataSet
                        'Return New WsasDataSet()

                    Case Is = WsAssemblyServiceType.DynamicDashboards
                    	'Return New WsasDynamicDashboards()

                    Case Is = WsAssemblyServiceType.FinanceCore
                        'Return New WsasFinanceCore()

                    Case Is = WsAssemblyServiceType.FinanceCustomCalculate
                        Return New finance_custom_calculate()

                    Case Is = WsAssemblyServiceType.FinanceGetDataCell
                        'Return New WsasFinanceGetDataCell()

                    Case Is = WsAssemblyServiceType.FinanceMemberLists
                        'Return New WsasFinanceMemberLists()

                    Case Is = WsAssemblyServiceType.SqlTableEditor
                        'Return New WsasSqlTableEditor()

                    Case Is = WsAssemblyServiceType.TableView
                        'Return New WsasTableView()

                    Case Is = WsAssemblyServiceType.XFBRString
                    	Return New xfbr_service()

                    Case Else
                        Return Nothing
                End Select

                Return Nothing
            Catch ex As Exception
                Throw New XFException(si, ex)
            End Try
        End Function

	End Class
End Namespace
